src/coord/src/coord/dataflow_builder.rs

```diff
@@ -290,175 +291,169 @@ impl<'a> DataflowBuilder<'a, mz_repr::Timestamp> {
 
         Ok(())
     }
+}
 
-    /// Prepares a relation expression for dataflow execution by preparing all
-    /// contained scalar expressions (see `prep_scalar_expr`) in the specified
-    /// style.
-    pub fn prep_relation_expr(
-        &self,
-        expr: &mut OptimizedMirRelationExpr,
-        style: ExprPrepStyle,
-    ) -> Result<(), CoordError> {
-        match style {
-            ExprPrepStyle::Index => {
-                expr.0.try_visit_mut_post(&mut |e| {
-                    // Carefully test filter expressions, which may represent temporal filters.
-                    if let MirRelationExpr::Filter { input, predicates } = &*e {
-                        let mfp =
-                            MapFilterProject::new(input.arity()).filter(predicates.iter().cloned());
-                        match mfp.into_plan() {
-                            Err(e) => coord_bail!("{:?}", e),
-                            Ok(_) => Ok(()),
-                        }
-                    } else {
-                        e.try_visit_scalars_mut1(&mut |s| self.prep_scalar_expr(s, style))
+/// Prepares a relation expression for dataflow execution by preparing all
+/// contained scalar expressions (see `prep_scalar_expr`) in the specified
+/// style.
+pub fn prep_relation_expr(
+    catalog: &CatalogState,
+    expr: &mut OptimizedMirRelationExpr,
+    style: ExprPrepStyle,
+) -> Result<(), CoordError> {
+    match style {
+        ExprPrepStyle::Index => {
+            expr.0.try_visit_mut_post(&mut |e| {
+                // Carefully test filter expressions, which may represent temporal filters.
+                if let MirRelationExpr::Filter { input, predicates } = &*e {
+                    let mfp =
+                        MapFilterProject::new(input.arity()).filter(predicates.iter().cloned());
+                    match mfp.into_plan() {
+                        Err(e) => coord_bail!("{:?}", e),
+                        Ok(_) => Ok(()),
                     }
-                })
-            }
-            ExprPrepStyle::OneShot { .. } => expr
-                .0
-                .try_visit_scalars_mut(&mut |s| self.prep_scalar_expr(s, style)),
+                } else {
+                    e.try_visit_scalars_mut1(&mut |s| prep_scalar_expr(catalog, s, style))
+                }
+            })
         }
+        ExprPrepStyle::OneShot { .. } => expr
+            .0
+            .try_visit_scalars_mut(&mut |s| prep_scalar_expr(catalog, s, style)),
     }
+}
 
-    /// Prepares a scalar expression for execution by handling unmaterializable
-    /// functions.
-    ///
-    /// Specifically, calls to unmaterializable functions are replaced if
-    /// `style` is `OneShot`. If `style` is `Index`, then an error is produced
-    /// if a call to a unmaterializable function is encountered.
-    pub fn prep_scalar_expr(
```

Comment:
these changes are hard to review due to the move, but the code is the same. GH does not do a good job at displaying it.